Scripts/detect_public_api_changes/compare.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+# -------------------------------------------------------------------
+# Setup & Info
+# -------------------------------------------------------------------
+
+BRANCH=$1
+REPO=$2
+TARGET="x86_64-apple-ios17.4-simulator"
+SDK="`xcrun --sdk iphonesimulator --show-sdk-path`"
+COMPARISON_VERSION_DIR_NAME="comparison_version_$BRANCH"
+DERIVED_DATA_PATH=".build"
+SDK_DUMP_INPUT_PATH="$DERIVED_DATA_PATH/Build/Products/Debug-iphonesimulator"
+ALL_TARGETS_LIBRARY_NAME="AdyenAllTargets"
+MODULE_NAMES=($(./Scripts/detect_public_api_changes/package_file_helper.swift Package.swift -print-available-targets))
+
+# -------------------------------------------------------------------
+# Convenience Functions
+# -------------------------------------------------------------------
+
+# Echos the entered information for validation
+function printRunInfo() {
+
+    echo "Branch: " $BRANCH
+    echo "Repo:   " $REPO
+    echo "Target: " $TARGET
+    echo "Dir:    " $COMPARISON_VERSION_DIR_NAME
+
+    echo "Modules:"
+    for MODULE in ${MODULE_NAMES[@]}; do
+        echo "-" $MODULE
+    done
+}
+
+# Clones the comparison project into a custom directory
+function setupComparisonRepo() {
+
+    rm -rf $COMPARISON_VERSION_DIR_NAME
+    mkdir $COMPARISON_VERSION_DIR_NAME
+    cd $COMPARISON_VERSION_DIR_NAME
+    git clone -b $BRANCH $REPO
+    cd ..
+}
+
+# Builds the project and temporarily modifies
+# some files to optimize the process
+#
+# $1 - The path to the current project dir
+#
+# Examples
+#
+#  buildProject . # If we're in the current project dir
+#  buildProject ../.. # If we're in the comparison project dir
+function buildProject() {
+
+    # Removing derived data if available
+    rm -rf .build
+
+    # We have to obscure the project file so `xcodebuild` uses the Package.swift to build the module
+    mv Adyen.xcodeproj Adyen.xcode_proj
+    
+    # Copying the Package.swift so we can revert the change done by the next step
+    cp Package.swift Package.sw_ift
+    
+    # Modify the Package.swift file to generate a product/library that contains all targets
+    # so we only have to build it once and can use it to diff all modules
+    $1/Scripts/detect_public_api_changes/package_file_helper.swift Package.swift -add-consolidated-library $ALL_TARGETS_LIBRARY_NAME
+
+    xcodebuild -scheme $ALL_TARGETS_LIBRARY_NAME \
+    -sdk $SDK \
```

Comment:
Just an FYI surrounding the use of an variable with quotes (f.e. `"$SDK"`) will make sure that when the variable is empty, contains spaces or special characters it doesn't break the script. In general it's better to use quotes, unless you intentionally don't need them